sched/clock/clock_gettime.c

```diff
@@ -131,6 +137,57 @@ int clock_gettime(clockid_t clock_id, struct timespec *tp)
         }
 #endif /* CONFIG_CLOCK_TIMEKEEPING */
     }
+#ifdef CONFIG_SCHED_CRITMONITOR
+  else if (clock_type == CLOCK_THREAD_CPUTIME_ID)
+    {
+      FAR struct tcb_s *tcb;
+
+      if (pid == 0)
+        {
+          /* Fetch the THREAD_CPUTIME for current thread */
+
+          tcb = nxsched_self();
+        }
+      else
+        {
+          tcb = nxsched_get_tcb(pid);
+        }
+
+      up_perf_convert(tcb->run_time, tp);
+    }
+  else if (clock_type == CLOCK_PROCESS_CPUTIME_ID)
+    {
+      FAR struct task_group_s *group;
+      unsigned long runtime;
+      irqstate_t flags;
+      int i;
+      FAR struct tcb_s *tcb;
+
+      if (pid == 0)
+        {
+          /* Fetch the PROCESS_CPUTIME for current process */
+
+          tcb = nxsched_self();
+        }
+      else
+        {
+           tcb = nxsched_get_tcb(pid);
+        }
+
```

Comment:
```suggestion
        {
          tcb = nxsched_get_tcb(pid);
        }

```

Reply:
fixed in https://github.com/apache/nuttx/pull/9701